server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java

```diff
@@ -1550,6 +1557,29 @@ private void scheduleIndexingAndPossibleDelete() {
                 });
             }
 
+            /**
+             * We must not close an index while it's being partially snapshotted; this counter tracks the number of ongoing
+             * close operations (positive) or partial snapshot operations (negative) in order to avoid them happening concurrently.
+             */
```

Comment:
Can we add a note here to say that full snapshot has no such issue because it retains index permits until the snapshot is created?

Could you please also remind me why we allow node restart once full snapshot starts? Does that mean the full snapshot can potentially fail?

Reply:
Sure, see 1021fc1.

> Could you please also remind me why we allow node restart once full snapshot starts? Does that mean the full snapshot can potentially fail?

That is a good question. Such a snapshot wouldn't fail, but it would come out as `PARTIAL`, and that would fail the test since we assert that full snapshots come out as `SUCCESS`. I've never seen it happen tho.